core/src/G3Frame.cxx

```diff
@@ -358,6 +361,14 @@ void G3Frame::blob_decode(struct blob_container &blob)
 	G3BufferInputStream is(*blob.blob);
 	cereal::PortableBinaryInputArchive item_ar(is);
 	item_ar >> make_nvp("val", ptr);
+
+#ifdef HAS_SUPERTIMESTREAM
+	// Convert to public type
+	auto v = std::dynamic_pointer_cast<G3SuperTimestream>(ptr);
+	if (!!v)
+		ptr = std::make_shared<G3TimestreamMap>(*v);
+#endif
+
```

Comment:
This is slightly ugly, but I don't see any other real way to do it. I might make the comment a little longer to explain to some future reader why any of this exists.

Reply:
I figured out a less-ugly solution by specializing cereal's `PolymorphicCasters::upcast()` function.  Seems to do the trick nicely.